### PR TITLE
fix(lspconfig): updating csharp-ls to look for sln when in a project structure

### PR DESCRIPTION
**Pull request by joro550**
Fixing an issue when loading a project in the lsp that it wasn't detecting the sln - it was only detecting the current csproj file

### CODE REVIEW

File: lua/lspconfig/server_configurations/csharp_ls.lua [hunk 1]
```diff
@@ -3,7 +3,12 @@ local util = require 'lspconfig.util'
 return {
   default_config = {
     cmd = { 'csharp-ls' },
-    root_dir = util.root_pattern('*.sln', '*.csproj', '*.fsproj', '.git'),
+    root_dir = function(fname)
+      return util.root_pattern '*.sln'(fname)
+        or util.root_pattern '*.csproj'(fname)
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/csharp_ls.lua:8]:
```suggestion
        or util.root_pattern('*.csproj', '*.fsproj', '.git')(fname)
```

**Comment by joro550** [comment]
@glepnir seems that suggestion had failed the lint 😅

File: lua/lspconfig/server_configurations/csharp_ls.lua [hunk 2]
```diff
@@ -3,7 +3,12 @@ local util = require 'lspconfig.util'
 return {
   default_config = {
     cmd = { 'csharp-ls' },
-    root_dir = util.root_pattern('*.sln', '*.csproj', '*.fsproj', '.git'),
+    root_dir = function(fname)
+      return util.root_pattern '*.sln'(fname)
+        or util.root_pattern('*.csproj', '*.fsproj', '.git')(fname)
+        or util.root_pattern '*.fsproj'(fname)
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/csharp_ls.lua:9]: these two lines can be removed.

**Comment by rudiejd** [comment]
@joro550  I think your most recent commit message is failing the commit linter😅 

I had the exact same issue and was about to open a PR to push my fix before I saw this, so I can do that if you don't have time

**Comment by joro550** [comment]
@rudiejd if you have time to invest please do it :smile: I haven't had time to fix this recently

**Pull request closed without merging**